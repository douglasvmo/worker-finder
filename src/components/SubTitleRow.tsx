import { PropsWithChildren } from "react";
import { StyleSheet, View, Text } from "react-native";
import RowTag from "./RowTag";
import { useTheme } from "@react-navigation/native";

type TProps = {
  title: string;
};

export default function SubTitleRow(props: PropsWithChildren<TProps>) {
  const { colors } = useTheme();
  return (
    <View style={style.row}>
      <RowTag />
      <Text style={[style.title, { color: colors.text }]}>{props.title}</Text>
      <View style={style.wrapper}>{props.children}</View>
    </View>
  );
}

const style = StyleSheet.create({
  row: {
    flexDirection: "row",
  },
  title: {
    fontFamily: "Inter-Bold",
    fontSize: 18,
  },
  wrapper: {
    marginLeft: "auto",
  },
});
