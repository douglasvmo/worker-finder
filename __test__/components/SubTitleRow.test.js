import React from "react";
import renderer from "react-test-renderer";
import SubTitleRow from "../../src/components/SubTitleRow";

describe("SubTitle Component", () => {
  test("render", () => {
    const tree = renderer
      .create(<SubTitleRow title="Cleaning Services" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
