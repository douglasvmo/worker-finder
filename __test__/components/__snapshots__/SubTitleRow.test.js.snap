// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubTitle Component render 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#CABDFF",
        "borderRadius": 4,
        "height": 20,
        "margin": 4,
        "width": 4,
      }
    }
  />
  <Text
    style={
      {
        "fontFamily": "Inter-Bold",
        "fontSize": 18,
      }
    }
  >
    Cleaning Services
  </Text>
  <View
    style={
      {
        "marginLeft": "auto",
      }
    }
  />
</View>
`;
